### ISSUE TITLE
Unexpected behavoir with ui.select and 'with_input=True'

### ISSUE THREAD

**Issue by thetableman** [zauberzeug/nicegui#2076]
### Description

Consider the example from the documentation of the `ui.select` with the `with_input=True` property set:

```py
from nicegui import ui

continents = [
    'Asia',
    'Africa',
    'Antarctica',
    'Europe',
    'Oceania',
    'North America',
    'South America',
]
ui.select(options=continents, with_input=True,
          on_change=lambda e: ui.notify(e.value)).classes('w-40')

ui.run()
```

When you type to search the expected behavior is correct, the list filters to the match the input. However, once you select an item the list remains filtered to the selected item value.

After selecting an item from the list of options, whether by filtering or not, the expected behavior is that the options return to the full un-filtered list so another selection can be made - as it does when `with_input` is not specified or is `False`.

This behavior may have changed after introducing changes in response to #2015.

**Comment by snowbollaanm** [comment]
Is there any progress on this bug? Providing the responses in #2015 I do agree with the initial change but don't believe the same functionality should be applied to a single select or by adding a parameter for it depending on the specific use case per each select.

**Comment by falkoschindler** [comment]
Thanks for the bug report, @thetableman!

And sorry for the late response. To be honest, I've been procrastinating quite a bit about this issue, since `ui.select` has become pretty complex with lots of subtle requirements and I needed some time to deep-dive into its implementation.

But it looks like I could fix the problem in PR #2175. It might be part of the next release.
